### ISSUE TITLE
Equality of Arrays behavior has changed.

### ISSUE THREAD

**Issue by gorzell** [etorreborre/specs2#540]
Hi,

I am not sure if it is related to #500, but the behavior of equality checks for Arrays appears to have changed recently.  It looks like it is doing reference comparison based on the output.

```
Array(31, 25, -116) != Array(31, 25, -116) /  expected: java.lang.String<31
25
-116> but was: java.lang.String<31
25
-116>
```

If this is intentional it probably warrants a minor version bump.

**Comment by etorreborre** [comment]
I think you are using `be_==` as a matcher which is indeed not doing the same check as before. As a workaround for now you can use `be_===` (typed equality) which should work fine.

**Comment by gorzell** [comment]
Thanks for the quick response.  We are using the `mustEqual` matcher which I believe gets translated as you mentioned.

It would be super helpful if in the future these kinds of behavior changes were reflected in the semantic versioning.

**Comment by gorzell** [comment]
I tried the typed form and it didn't seem to work in all cases.

**Comment by etorreborre** [comment]
This was not intended as a behaviour change. I think that new features must not come at the price at breaking the stability of your test suite by now. I am publishing a fix right now.

**Comment by gorzell** [comment]
Awesome.  Really appreciate the quick response.

**Comment by etorreborre** [comment]
The fix is available in `3.8.7-20170202221116-f32a298`.
